operators/constellation-node-operator/internal/cloud/aws/client/scalinggroup.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/autoscaling"
+	scalingtypes "github.com/aws/aws-sdk-go-v2/service/autoscaling/types"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+)
+
+// GetScalingGroupImage returns the image URI of the scaling group.
+func (c *Client) GetScalingGroupImage(ctx context.Context, scalingGroupID string) (string, error) {
+	launchTemplate, err := c.getScalingGroupTemplate(ctx, scalingGroupID)
+	if err != nil {
+		return "", err
+	}
+
+	if launchTemplate.LaunchTemplateData == nil {
+		return "", fmt.Errorf("launch template data is nil for scaling group %q", scalingGroupID)
+	}
+
+	if launchTemplate.LaunchTemplateData.ImageId == nil {
+		return "", fmt.Errorf("image ID is nil for scaling group %q", scalingGroupID)
+	}
+
+	return *launchTemplate.LaunchTemplateData.ImageId, nil
+}
+
+// SetScalingGroupImage sets the image URI of the scaling group.
+func (c *Client) SetScalingGroupImage(ctx context.Context, scalingGroupID, imageURI string) error {
+	launchTemplate, err := c.getScalingGroupTemplate(ctx, scalingGroupID)
+	if err != nil {
+		return fmt.Errorf("failed to get launch template for scaling group %q: %w", scalingGroupID, err)
+	}
+
+	_, err = c.ec2Client.CreateLaunchTemplateVersion(
+		ctx,
+		&ec2.CreateLaunchTemplateVersionInput{
+			LaunchTemplateData: &ec2types.RequestLaunchTemplateData{
+				ImageId: &imageURI,
+			},
+			LaunchTemplateId: launchTemplate.LaunchTemplateId,
+			SourceVersion:    toPtr(fmt.Sprintf("%d", *launchTemplate.VersionNumber)),
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create launch template version: %w", err)
+	}
+
+	// set created version as default
+	_, err = c.ec2Client.ModifyLaunchTemplate(
+		ctx,
+		&ec2.ModifyLaunchTemplateInput{
+			LaunchTemplateId: launchTemplate.LaunchTemplateId,
+			DefaultVersion:   toPtr(fmt.Sprintf("%d", *launchTemplate.VersionNumber+1)),
```

Comment:
`VersionNumber` could be nil

Reply:
Also: wouldn't it be a bit safer to take the new DefaultVersion from the return value of the previous `CreateLaunchTemplateVersion`?